File: AIDatabaseAnalyzer/Models/AppModels.cs
using CommunityToolkit.Mvvm.ComponentModel;

namespace AIDatabaseAnalyzer.Models;

public enum AnalysisStatus { Pending, Processing, Completed, Error }

public partial class AnalysisTopic : ObservableObject
{
    [ObservableProperty] private string title;
    [ObservableProperty] private string prompt;
    [ObservableProperty] private bool isSelected = true;
    [ObservableProperty] private bool isCustom = false;
    [ObservableProperty] private AnalysisStatus status = AnalysisStatus.Pending;
}

public partial class ComponentSelection : ObservableObject
{
    [ObservableProperty] private string name;
    [ObservableProperty] private bool isSelected = true;
}

public partial class TableSelection : ObservableObject
{
    [ObservableProperty] private string name;
    [ObservableProperty] private bool isSelected = true;
}

File: flutter_app/lib/models/app_models.dart
enum AnalysisStatus { pending, processing, completed, error }

class AnalysisTopic {
  final String title;
  final String prompt;
  bool isSelected;
  bool isCustom;
  AnalysisStatus status;

  AnalysisTopic({
    required this.title,
    required this.prompt,
    this.isSelected = true,
    this.isCustom = false,
    this.status = AnalysisStatus.pending,
  });
}

class ComponentSelection {
  final String name;
  bool isSelected;

  ComponentSelection({required this.name, this.isSelected = true});
}

class TableSelection {
  final String name;
  bool isSelected;

  TableSelection({required this.name, this.isSelected = true});
}

class ReportSection {
  final String title;
  final String content;

  ReportSection({required this.title, required this.content});
}

class SchemaFilterConfig {
  final List<String> selectedTables;
  final bool includeViews;
  final bool includeIndexes;
  final bool includeTriggers;
  final bool includeSps;
  final bool includeConstraints;
  final bool includeSampleData;

  SchemaFilterConfig({
    required this.selectedTables,
    required this.includeViews,
    required this.includeIndexes,
    required this.includeTriggers,
    required this.includeSps,
    required this.includeConstraints,
    required this.includeSampleData,
  });
}
